libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -337,6 +373,10 @@ void AP_Baro::init(void)
     case AP_BoardConfig::PX4_BOARD_PH2SLIM:
         ADD_BACKEND(AP_Baro_MS56XX::probe(*this,
                                           std::move(hal.spi->get_device(HAL_BARO_MS5611_NAME))));
+#if APM_BUILD_TYPE(APM_BUILD_ArduSub)
+		ADD_BACKEND(AP_Baro_MS56XX::probe(*this,
+										  std::move(hal.i2c_mgr->get_device(HAL_BARO_MS5837_BUS, HAL_BARO_MS5837_I2C_ADDR))));
```

Comment:
unfortunately we can't just probe all buses as MS used the same I2C ID and same detection method (same CRC on rom) for different sensor types. There is no way to tell between a MS airspeed sensor and a baro for example.